libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -3079,7 +3084,23 @@ void IRGeneratorForStatements::writeToLValue(IRLValue const& _lvalue, IRVariable
 					offsetArgument <<
 					_value.commaSeparatedListPrefixed() <<
 					")\n";
+			},
```

Comment:
```c++
auto storageVisitor = [&](auto const& _storage, Location _location) {
     ... code ...
};

std::visit(
  util::GenericVisitor{
    [&](IRLValue::Storage const& _storage) { storageVisitor(_storage, Location::Unspecified); },
    [&](IRLValue::TransinetStorage const& _storage) { storageVisitor(_storage, Location::Transient); }
  }
);
```

Reply:
This one seems still not done.

And we should really do the same thing in `readFromLValue()`.

Reply:
Yeah, I didn't do that one on purpose since I was hoping to have more discussion around it. I think in this case, the code duplication is preferable.

Reply:
But are there any downsides really? Seems like in these cases the lambda would be straightforward and it's only one value that needs to be a parameter. The other things are just names.

Reply:
All of this is subjective, so for what it's worth, I find that this approach deteriorates readability (and flow) just for the sake of minor de-duplication of code. I'll also need to have 3 such lambdas for 3 different functions. I'eave it as is because I find it nicer that way, and if it turns out as a showstopper for Daniel or yourself, feel free to slap a CR, and I'll change it.